src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,72 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+import duke.DukeException;
+import duke.util.Parser;
+
+
+/**
+ * Task with a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDate by;
```

Comment:
Hello Wei Quan, the field `by` could be renamed to the noun `dueDate` to better explain what it represents. Otherwise, `by` could also potentially refer to the person who added this task.

```suggestion
    private LocalDate dueDate;
```